geniza/corpus/models.py

```diff
@@ -234,7 +234,8 @@ def shelfmark(self):
         # access via textblock so we follow specified order,
         # use dict keys to ensure unique
         return ' + '.join(dict.fromkeys(block.fragment.shelfmark
-                          for block in self.textblock_set.all()))
+                          for block in self.textblock_set.all()
```

Comment:
suggest filtering on certain=true on the textblock queryset instead of the extra test

Reply:
Please double check that I understood you correctly: a68757b. If it looks good to you, feel free to merge!